app/src/test/java/com/ikarabulut/BoardTest.java

```diff
@@ -69,4 +72,137 @@ void setRandomGrid() {
+            int numOfAliveNeighbors = board.getNumberOfAliveNeighbors(1,1);
+            assertEquals(numOfAliveNeighbors, 0);
+        }
+
+        @Test
+        @DisplayName("A cell with 1 alive neighbor should return 1 alive neighbor")
+        void getNumberOfAliveNeighbors_When1NeighborIsAlive_Returns1() {
+            board.setAliveAt(0,1);
+
+            int numOfAliveNeighbors = board.getNumberOfAliveNeighbors(1, 1);
+
+            assertEquals(numOfAliveNeighbors, 1);
+        }
+
+        @Test
+        @DisplayName("A cell with 2 alive neighbors should return 2 alive neighbors")
+        void getNumberOfAliveNeighbors_When2NeighborsAreAlive_Returns2() {
+            board.setAliveAt(0,1);
+            board.setAliveAt(0,2);
+
+            int numOfAliveNeighbors = board.getNumberOfAliveNeighbors(1,1);
+
+            assertEquals(numOfAliveNeighbors, 2);
+        }
+
+        @Test
+        @DisplayName("A cell with 8 alive neighbors should return 8 alive neighbors")
+        void getNumberOfAliveNeighbors_WhenAllNeighborsAreAlive_Returns8() {
+            setAliveBoard();
+
+            int numOfAliveNeighbors = board.getNumberOfAliveNeighbors(1, 1);
+
+            assertEquals(numOfAliveNeighbors, 8);
+        }
+
+        @Test
+        @DisplayName("Should count neighbors that wrap from the right side of a row to the left side")
+        void getNumberOfAliveNeighbors_WhenCellIsOnRightEdge_ReturnsAliveCellsFromLeftEdge() {
+            Board largerBoard = new Board(5,5);
+            setDeadBoard();
+            largerBoard.setAliveAt(2, 0);
+
+            int numOfAliveNeighbors = largerBoard.getNumberOfAliveNeighbors(2, 4);
+
+            assertEquals(numOfAliveNeighbors, 1);
+        }
+
+        @Test
+        @DisplayName("Should count neighbors that wrap diagonally from the top left to the bottom right of the board")
+        void getNumberOfAliveNeighbors_WhenCellIsBottomRightEdge_ReturnsAliveCellsFromTopRightEdge() {
+            Board largerBoard = new Board(5, 5);
+            setDeadBoard();
+            largerBoard.setAliveAt(0,0);
+
+            int numOfAliveNeighbors = largerBoard.getNumberOfAliveNeighbors(4, 4);
+
+            assertEquals(numOfAliveNeighbors, 1);
+        }
+
+        @Test
+        @DisplayName("Should count neighbors that wrap from the top of a column to the bottom")
+        void getNumberOfAliveNeighbors_WhenCellIsOnBottomEdge_ReturnsAliveNeighborsFromTop() {
+            Board largerBoard = new Board(5, 5);
+            setDeadBoard();
+            largerBoard.setAliveAt(4, 2);
+
+            int numOfAliveNeighbors = largerBoard.getNumberOfAliveNeighbors(0, 2);
+
+            assertEquals(numOfAliveNeighbors, 1);
+        }
+
+        @Test
+        @DisplayName("Should count neighbors that wrap from the top right of a column to the bottom left")
+        void getNumberOfAliveNeighbors_WhenCellIsTopRight_ReturnsAliveNeighborsFromBottomLeftEdge() {
```

Comment:
I notice we're testing that the top edge wraps to the bottom edge. Are we also testing that the bottom edge wraps to the top?  I know world wrapping is the trickiest bit, so it might help to have tighter test coverage around that algorithm.

Reply:
I am testing both corners but there are a few more cases that could be tested.